pkg/trace/stats/concentrator.go

```diff
@@ -308,8 +310,12 @@ func (c *Concentrator) flushNow(now int64, force bool) *pb.StatsPayload {
 			ImageTag:     k.ImageTag,
 			Stats:        s,
 		}
+		if k.ContainerTags != "" {
+			p.Tags = append(p.Tags, strings.Split(k.ContainerTags, ",")...)
```

Comment:
Hmm I was hoping that here we could simply do `p.Tags = tags`, but I'm not sure where `tags` would be stored. I don't want us to modify the `ClientStatsBucket` which is the type of `s`, and I know we can't add a slice of strings to `PayloadAggregationKey` which is the type of `k`.

I'm going to see if there's another way. Perhaps we have to make do with the string joining and splitting after all.

Reply:
What if we create a `map[string][]string` in the [RawBucket](https://github.com/DataDog/datadog-agent/blob/c2a48f519aa4231ba751409f4fb0eaa912b01bc4/pkg/trace/stats/statsraw.go#L101)?

We'd have a map of containerID -> containerTags for each bucket separately.
During the exporting of the bucket into stats, we'd make [a new temporary map of containerID -> containerTags below the existing map we make](https://github.com/DataDog/datadog-agent/blob/cec4dfe545f33e57d503c0c4ee685878794ab310/pkg/trace/stats/concentrator.go#L272).

Finally below we'd then do:
```
		p := &pb.ClientStatsPayload{
			Env:          k.Env,
             ...
			Stats:        s,
            Tags: cidToTags[k.ContainerID],
		}
```

This way we:
1. Avoid all string join/split operations.
2. "Cache" at the bucket level just once rather than potentially writing the same joined strings for several different [AggregationKey](https://github.com/DataDog/datadog-agent/blob/c2a48f519aa4231ba751409f4fb0eaa912b01bc4/pkg/trace/stats/aggregation.go#L27) in the same raw bucket.
3. Use the latest mapping of container ID -> container tags. This is more relevant for the OTel case where we do resolution first and pass in the result. This is less relevant for the canonical APM stats path where we resolve container tags at the very last possible moment before writing data out.

What do you think @songy23 @ajgajg1134 @dineshg13 ?

Reply:
Agreed with the benefits @jdgumz listed with having the tags in RawBucket, I also do not feel it is a great solution to keep the container tags in aggregation key. I moved it to RawBucket.

It does not improve the performance _significantly_ though, due to multiple intermediate temp map being allocated. Benchmark results:

Old code without https://github.com/DataDog/datadog-agent/pull/23503/commits/9eb0dee8369803aa13ee2c188f6c59f04b450267
```
% go test -bench="BenchmarkOTelContainerTags"  -benchtime=1000000x -benchmem
goos: darwin
goarch: arm64
pkg: github.com/DataDog/datadog-agent/pkg/trace/stats
BenchmarkOTelContainerTags-10    	 1000000	      6703 ns/op	    7032 B/op	      74 allocs/op
PASS
ok  	github.com/DataDog/datadog-agent/pkg/trace/stats	7.191s
```

New code with https://github.com/DataDog/datadog-agent/pull/23503/commits/9eb0dee8369803aa13ee2c188f6c59f04b450267
```
% go test -bench="BenchmarkOTelContainerTags"  -benchtime=1000000x -benchmem
goos: darwin
goarch: arm64
pkg: github.com/DataDog/datadog-agent/pkg/trace/stats
BenchmarkOTelContainerTags-10    	 1000000	      6349 ns/op	    7136 B/op	      73 allocs/op
PASS
ok  	github.com/DataDog/datadog-agent/pkg/trace/stats	6.856s
```